tests/backend/integration/api/notifications/test_resources.py

```diff
@@ -0,0 +1,237 @@
+            self.url,
+            headers={"Authorization": self.test_receiver_token},
+        )
+        response_body = response.get_json()
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response_body["subject"], TEST_SUBJECT)
+        self.assertEqual(response_body["message"], TEST_MESSAGE)
+        self.assertEqual(response_body["fromUsername"], self.test_sender.username)
+
+    def test_delete_message_returns_401(self):
+        """
+        Test that endpoint returns 401 when an unauthenticated user deletes a message
+        """
+        response = self.client.delete(self.url)
+        response_body = response.get_json()
+        self.assertEqual(response.status_code, 401)
+        self.assertEqual(response_body["SubCode"], "InvalidToken")
+
+    def test_delete_message_returns_403(self):
+        """
+        Test that endpoint returns 403 when a user deletes another user's message
+        """
+        response = self.client.delete(
+            self.url,
+            headers={"Authorization": self.test_sender_token},
+        )
+        response_body = response.get_json()
+        self.assertEqual(response.status_code, 403)
+        self.assertEqual(response_body["SubCode"], "AccessOtherUserMessage")
+
+    def test_delete_message_returns_404(self):
+        """
+        Test that endpoint returns 404 when a user deletes a non-existent message
+        """
+        response = self.client.delete(
+            self.non_existent_url,
+            headers={"Authorization": self.test_sender_token},
+        )
+        response_body = response.get_json()
+        self.assertEqual(response.status_code, 404)
+        self.assertEqual(response_body["Error"], MESSAGES_NOT_FOUND)
+        self.assertEqual(response_body["SubCode"], NOT_FOUND)
+
+    def test_delete_message_returns_200(self):
+        """
+        Test that endpoint returns 200 when a user successfully deletes their existent message
+        """
+        response = self.client.delete(
+            self.url,
+            headers={"Authorization": self.test_receiver_token},
+        )
+        response_body = response.get_json()
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response_body["Success"], "Message deleted")
+
+
+class TestNotificationsAllAPI(BaseTestCase):
+    def setUp(self):
+        super().setUp()
+        self.test_user = create_canned_user()
+        self.test_user_token = generate_encoded_token(self.test_user.id)
+        self.url = "/api/v2/notifications/"
+
+    def test_get_message_notifications_returns_401(self):
+        """
+        Test that endpoint returns 401 when an unauthenticated user accesses messsage notifications
+        """
+        response = self.client.get(self.url)
+        response_body = response.get_json()
+        self.assertEqual(response.status_code, 401)
+        self.assertEqual(response_body["SubCode"], "InvalidToken")
+
+    def test_get_messages_returns_200(self):
```

Comment:
Also, test all of the query parameters to ensure that they all have the expected impact on the response.